### DISCUSSION TITLE
Более очевидные disabled значения прозрачности компонента Button

### DISCUSSION THREAD

**Discussion by some-dudes-work-account** [VKCOM/VKUI#5979] [Ideas]
Не уверен, что это подходящее место для обсуждения дизайна VKUI, но не могу придумать где было бы более уместно.

Когда кнопка содержит текст и стоит рядом с другими кнопками, когда у них есть не прозрачный бэкграунд, у меня нет претензий к их disabled состояниям. Однако, если без фона, если есть только иконки или кнопка это часть тулбара из нескольких ~~тернарных~~ третичных кнопок, где все они имеют только иконку, во многих комбинациях disabled состояние визуально сложно отличимо от активных кнопок.

Я собрал демку https://stackblitz.com/edit/vitejs-vite-5goobx?file=src%2FToolbar.tsx,src%2FApp.tsx&terminal=dev

Например, какие из кнопок здесь не кликабельны? 

![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (1)](https://github.com/VKCOM/VKUI/assets/100410177/908344a9-4e06-47d1-ac86-6ea398bbb225)

<details>
  <summary>больше примеров состояний где оно выглядит недостаточно контрастно</summary>

![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (2)](https://github.com/VKCOM/VKUI/assets/100410177/7df4422f-f24e-4571-8fd3-b3551c0485a5)

![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (3)](https://github.com/VKCOM/VKUI/assets/100410177/606cc9ae-bd85-4c73-889b-d8b913fab50d)

![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (4)](https://github.com/VKCOM/VKUI/assets/100410177/6b948176-7927-479d-a90e-208a0d7b499f)
</details>

<details>
  <summary>Примеры состояний где оно выглядит хорошо</summary>
  
  ![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (7)](https://github.com/VKCOM/VKUI/assets/100410177/20dadc36-0646-41ca-accb-f3b814542a9e)

  ![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (5)](https://github.com/VKCOM/VKUI/assets/100410177/7dcf01f3-5514-4098-b83c-4091988abb83)

  ![vitejsvite5goobx-cs2m--5173--2a8baea5 local-credentialless webcontainer io_ (6)](https://github.com/VKCOM/VKUI/assets/100410177/267a1c6c-5c53-4aa1-aa18-0468cabce892)
</details>

Я вижу, что значение переопределяется в 0.64 с дефолтных 0.4 для большинства из комбинаций, кроме link.

https://github.com/VKCOM/VKUI/blob/bc8cd428f120514bd60521f8f704adc2d5b97a41/packages/vkui/src/components/Button/Button.module.css#L69-L80

Может быть можно к селекторам дописать `:not(.Button--singleIcon)`, чтобы оно не сливалось так сильно? 

Или отдельно сами иконки в before и after слотах таргетить и делать дополнительно более прозрачными когда сама кнопка 0.64?

**Comment by inomdzhon** [comment]
Спасибо за предложение 🙏 

Закинул @VKCOM/vkui-design
